packages/api/cms-api/src/blocks/YouTubeVideoBlock/you-tube-video.block.ts

```diff
@@ -0,0 +1,34 @@
+import { BlockDataInterface, BlockField, createBlock, inputToData, typesafeMigrationPipe } from "@comet/blocks-api";
+import { IsOptional, IsString, Matches } from "class-validator";
+
+import { BaseVideoBlockData, BaseVideoBlockInput } from "../BaseVideoBlockData";
+import { RemoveAspectRatioMigration } from "./migrations/1-remove-aspect-ratio.migration";
+import { AddPreviewImageMigration } from "./migrations/2-add-preview-image.migration";
+
+class YouTubeVideoBlockData extends BaseVideoBlockData {
+    @BlockField({ nullable: true })
+    youtubeIdentifier?: string;
+}
+
+class YouTubeVideoBlockInput extends BaseVideoBlockInput {
+    @IsOptional()
+    @IsString()
+    @BlockField({ nullable: true })
+    // regex from https://stackoverflow.com/a/51870158
+    @Matches(
+        /(https?:\/\/)?(((m|www)\.)?(youtube(-nocookie)?|youtube.googleapis)\.com.*(v\/|v=|vi=|vi\/|e\/|embed\/|user\/.*\/u\/\d+\/)|youtu\.be\/)([_0-9a-zA-Z-]+)/,
+    )
```

Comment:
This was recently fixed in main (#2260), we should merge main before merging this.

Reply:
oh.
when will this be done?